bin/proof-generator-multithreaded/src/main.cpp

```diff
@@ -137,10 +137,11 @@ struct prover {
             context_.find<nil::proof_generator::aspects::prover_vanilla>()->input_assignment_file_path();
 
         skip_verification = context_.find<nil::proof_generator::aspects::prover_vanilla>()->is_skip_verification_mode_on();
+        shard0_mem_scale = context_.find<nil::proof_generator::aspects::prover_vanilla>()->get_shard0_mem_scale();
 
         proof_file = context_.find<nil::proof_generator::aspects::prover_vanilla>()->output_proof_file_path();
 
-        std::vector<std::string> arguments = { "program_name", "--shard0-mem-scale", "4800" };
+        std::vector<std::string> arguments = { "Program_name", "--shard0-mem-scale", std::to_string(shard0_mem_scale) };
```

Comment:
We don't care about the program name here, it's not used anyway, so just kept this fake parameter here.